snowplow_tracker/events/self_describing.py

```diff
@@ -0,0 +1,91 @@
+from snowplow_tracker.events.event import Event
+from snowplow_tracker import SelfDescribingJson
+
+# """
+#     self_describing.py
+
+#     Copyright (c) 2013-2023 Snowplow Analytics Ltd. All rights reserved.
+
+#     This program is licensed to you under the Apache License Version 2.0,
+#     and you may not use this file except in compliance with the Apache License
+#     Version 2.0. You may obtain a copy of the Apache License Version 2.0 at
+#     http://www.apache.org/licenses/LICENSE-2.0.
+
+#     Unless required by applicable law or agreed to in writing,
+#     software distributed under the Apache License Version 2.0 is distributed on
+#     an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+#     express or implied. See the Apache License Version 2.0 for the specific
+#     language governing permissions and limitations there under.
+# """
+from typing import Optional
+from snowplow_tracker.typing import JsonEncoderFunction
+
+from snowplow_tracker.constants import UNSTRUCT_EVENT_SCHEMA
+
+
+class SelfDescribing(Event):
+    def __init__(
+        self,
+        event_json: SelfDescribingJson,
+        encode_base64: bool,
+        json_encoder: Optional[JsonEncoderFunction],
```

Comment:
This is a bit unintuitive because the `encode_base64` and `json_encoder` are configurations in the `Tracker` instance and this would require the user to pass them twice – once when creating the tracker and once when creating this event.

My suggestion would be to override the `build_payload` method from the `Event` class in this `SelfDescribing` class. That one receives both of the configurations from the tracker so we could serialize the JSON and encode it at that point.